blog/2022-04-12-ephemeral-containers.md

```diff
@@ -0,0 +1,217 @@
+
+If PSP is your only policy tool and you are planning to upgrade to v1.23, don't. As PSP is deprecated no new features have been added, and that includes policy enforcement on ephemeral containers. Which means that any security context in an ephemeral container is allowed no matter the PSP in the cluster.  The PSP below will have no affect when adding an ephemeral container to a Pod which is privileged.
+
+```yaml
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: default
+spec:
+  privileged: false
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  runAsUser:
+    rule: RunAsAny
+  fsGroup:
+    rule: RunAsAny
+  volumes:
+  - '*'
+```
+
+### RBAC
+
+Disallowing ephemeral containers with RBAC could be an option if the feature is not needed and it is not possible to disable the feature completely. The [KEP-277: Ephemeral Containers](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/277-ephemeral-containers/README.md) state the following about using RBAC to disable ephemeral containers.
+
+> Cluster administrators will be expected to choose from one of the following mechanisms for restricting usage of ephemeral containers:
+> * Use RBAC to control which users are allowed to access the /ephemeralcontainers subresource.
+> * Write or use a third-party admission controller to allow or reject Pod updates that modify ephemeral containers based on the content of the update.
+> * Disable the feature using the EphemeralContainers feature gate.
+
+RBAC is additive which means that it is not possible to remove permissions from a role. This type of mitigation obviously does not matter if all users a cluster admin, which they should not be, so we assume that new roles are created for the cluster consumers. In this case having a look at the existing roles can be enough to make sure that the subresource is not included in the role.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: edit
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - pods/attach
+  - pods/exec
+  - pods/portforward
+  - pods/proxy
+  verbs:
+  - create
+  - delete
+  - deletecollection
+  - patch
+  - update
+```
+
+## Checking Policy Enforcement
+
+So what if we have a cluster with ephemeral containers enabled, and you are unsure if the correct policies are enforced? Or you just want to verify you work. The debug command does not expose any options to set any security context configuration, so we need another option. Ephemeral containers cannot be defined in a Pod when it is created neither can it be added with an update. The Kubernetes documentation states the following.
```

Comment:
Not sure what you want to say with "verify you work"? Should it be "verify that ephemeral containers work"?
"when it is created *and* neither can it ..."